### PR TITLE
Set up JSON response infrastructure

### LINKED ISSUE #213: Configure requester functions to accept and invoke "promise" callbacks

### LINKED ISSUE #221: Route to new page on successful login

### PR DESCRIPTION
**Pull request by warrenshen**
Fix #213 
Fix #221

### CODE REVIEW

File: app/assets/javascripts/components/authentication/login_form.es6.jsx [hunk 1]
```diff
@@ -51,6 +51,7 @@ class LoginForm extends Component {
   }
 
   render() {
+    console.log(this.state.errors);
```
Comment (warrenshen) [inline app/assets/javascripts/components/authentication/login_form.es6.jsx]: Leaving this `console.log` here for reference for @unzunz!

File: app/assets/javascripts/components/authentication/login_form.es6.jsx [hunk 2]
```diff
@@ -24,15 +25,15 @@ class LoginForm extends Component {
...
-      }
-    );
+    var params = {
+      user: {
+        email: this.state.email,
+        password: this.state.password,
+      },
+    };
+    var resolve = (response) => { window.location = RouteConstants.students.index };
+    var reject = (response) => { this.setState({ errors: true }) };
```
Comment (warrenshen) [inline app/assets/javascripts/components/authentication/login_form.es6.jsx]: We're going to be introducing something called the "rocket", which is a nice way of creating functions without having to use '.bind(this)' (I mentioned this earlier and here it is!).

File: app/assets/javascripts/components/authentication/login_form.es6.jsx (hunk 1 above)
Comment (unzunz) [inline app/assets/javascripts/components/authentication/login_form.es6.jsx]: :+1:

File: app/assets/javascripts/components/authentication/login_form.es6.jsx (hunk 2 above)
Comment (maxwolffe) [inline app/assets/javascripts/components/authentication/login_form.es6.jsx]: Noice.

**Comment by maxwolffe** [comment]
LGTM

**Pull request merged**